### ISSUE TITLE
Variational posteriors distribution of latent variable z_k in CLUE

### ISSUE THREAD

**Issue by bestadcarry** [gao-lab/GLUE#67]
Hi, 

I have a question regarding [CLUE](https://xinmingtu.cn/assets/pdf/clue_paper.pdf) that I was hoping you could help me with. While reading the paper, I came across equation (6) of CLUE, which suggests that the embedding z_k is sampled from the product of expert. When each expert (cross-encoder) is Gaussian, then the product is also Gaussian, and z_k should be sampled from this product distribution. However, when I reviewed the [code](https://github.com/gao-lab/GLUE/blob/73c49acc1209618edc40815ff53f8497a9c7d5ec/scglue/models/scclue.py#L742), I found that each embedding is generated from a single modality and then combined by average. I'm wondering if I might have missed something or misunderstood CLUE.

I would be grateful if you could offer any hints or suggestions to help me better understand this. Thank you very much for your time and help!

**Comment by Jeff1995** [comment]
Hi @bestadcarry! You are right. The product-of-expert distribution should actually be sampled from a precision-weighted average Gaussian, instead of a simple average of independent samples. The CLUE code was initially written during the NeurIPS 2021 competition where we used unweighted average as a rough approximation. After the competition however, we did some additional tests with a proper product-of-expert implementation, but did not find much of a difference in terms of integration accuracy, so the public code was not changed. Sorry for the confusion!
